PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -721,7 +721,18 @@ bool FileManager::reloadBuffer(BufferID id)
 	buf->setLoadedDirty(false);	// Since the buffer will be reloaded from the disk, and it will be clean (not dirty), we can set _isLoadedDirty false safetly.
 								// Set _isLoadedDirty false before calling "_pscratchTilla->execute(SCI_CLEARALL);" in loadFileData() to avoid setDirty in SCN_SAVEPOINTREACHED / SCN_SAVEPOINTLEFT
 
+	bool bUnsyncStateBeforeReload = buf->isUnsync(); // store for a possible unsync state restoration later 
+	buf->setUnsync(false);	// we have to ensure that isUnsync is false before we try to call the loadFileData below for the reloading
+							// - after a successful reload, the source file and its filebuffer in N++ are the same, so the isDirty should be false (and the isUnsync too)
+							// - but when the isUnsync is true at this point, any successfully reloaded filebuffer later will end up marked as dirty!
+							//   (similar reason as in the above setLoadedDirty(false), but now it is the "_pscratchTilla->execute(SCI_SETSAVEPOINT)" in loadFileData(),
+							//    which ends up in the Notepad_plus::notify(), case SCN_SAVEPOINTLEFT, where is logic "if (buf->isUnsync()) isDirty = true")

```

Comment:
It's nice to have comment, but it's kind of too verbose.
I would give a simple explanation then provide the URL of issue.

Reply:
Ok, I will shorten that.